src/Controllers/OAuthController.php

```diff
@@ -55,6 +57,13 @@ public function callBack(): RedirectResponse
         }
         $allGet = $this->request->getGet();
 
+        // if permission is denied or was cancelled by user.
+        if (isset($allGet['error']) && $allGet['error'] === self::ACCESS_DENIED) {
+            $oauthName = ucwords(session('oauth_name'));
```

Comment:
We have already translated the driver names for other languages. So you should use these to display drivers.
`lang('ShieldOAuthLang.Github.github')`, `lang('ShieldOAuthLang.Google.google')`, `lang('ShieldOAuthLang.Yahoo.yahoo')`

Reply:
Sample used:
https://github.com/datamweb/shield-oauth/blob/1f36be95119a3fb88628332220643f9973a9a522/src/Libraries/Basic/ShieldOAuth.php#L137-L140

Reply:
Could you please verify if this way of accessing driver names is correct?

```php
$oauth  = session('oauth_name');
$oauthh = ucwords($oauth);

$oauthName = lang("ShieldOAuthLang.{$oauthh}.{$oauth}");
```